yt_dlp/extractor/facebook.py

```diff
@@ -493,14 +509,19 @@ def process_formats(info):
 
         def extract_relay_data(_filter):
             return self._parse_json(self._search_regex(
-                r'handleWithCustomApplyEach\([^,]+,\s*({.*?%s.*?})\);' % _filter,
+                r'data-sjs>({.*?%s.*?})</script>' % _filter,
                 webpage, 'replay data', default='{}'), video_id, fatal=False) or {}
 
         def extract_relay_prefetched_data(_filter):
             replay_data = extract_relay_data(_filter)
             for require in (replay_data.get('require') or []):
                 if require[0] == 'RelayPrefetchedStreamCache':
                     return try_get(require, lambda x: x[3][1]['__bbox']['result']['data'], dict) or {}
+                else:
+                    data = traverse_obj(require, (
+                        ..., ..., '__bbox', 'require', lambda _, v: "RelayPrefetchedStreamCache" in v,
+                        ..., ..., "__bbox", "result", "data"))
+                    return data[0] if data else {}
```

Comment:
```suggestion
        def extract_relay_prefetched_data(_filter):
            return traverse_obj(extract_relay_data(_filter), (
                'require', (None, (..., ..., ..., '__bbox', 'require')),
                lambda _, v: 'RelayPrefetchedStreamCache' in v, ..., ...,
                '__bbox', 'result', 'data', {dict}), get_all=False) or {}
```

Reply:
Cool, but that's just a merge of those. Is there some equivalent of xpath `//require[...]`? (For future reference)

Reply:
afaik no